### PR TITLE
Babel 7 migration

### LINKED ISSUE #1734: Migrate to babel@7

### PR DESCRIPTION
**Pull request by vikr01**
Resolves #1734

- [x] Update all babel dependencies to the latest version
- [ ] Make `./app` recognize root `babel` config and presets/plugins
- [x] Remove any deprecated `babel` plugins
- [x] Change all instances of `babel-register` to `@babel/register`
- [x] Add `babel-core@7.0.0-bridge.0` for `jest`
- [x] Update any webpack files including `babel` plugins/presets
- [x] Utilize the new `development` mode for `@babel/preset-react`
- [x] Add `@babel/preset-flow` separately since it is no longer included with `@babel/preset-react`
- [x] Remove any deprecated `babel` presets
  - [x] Remove `babel-preset-stage-0` in favor of including all `@babel/plugin-proposal-*`
  - [x] Remove `babel-preset-react-optimize` in favor of including updated versions of the plugins bundled in it
- [x] Change all `require(module)` statements importing from an es6 export to `require(module).default`
  - [ ] Add a note about needing to change `require` statements for any users upgrading `electron-react-boilerplate`
- [ ] Test and remove any `babel` plugins that break `babel` 7

### CODE REVIEW

File: app/store/configureStore.js [hunk 1]
```diff
@@ -1,6 +1,12 @@
 // @flow
-if (process.env.NODE_ENV === 'production') {
-  module.exports = require('./configureStore.prod'); // eslint-disable-line global-require
-} else {
-  module.exports = require('./configureStore.dev'); // eslint-disable-line global-require
-}
+import configureStoreDev from './configureStore.dev';
```
Comment (vikr01) [inline app/store/configureStore.js]: Will `webpack` remove this when "tree-shaking" for production?

This does add an unnecessary step of importing both modules first then deciding what to do with them.

Comment (amilajack) [reply app/store/configureStore.js]: Yea, webpack will remove it. We are using `EnvironmentPlugin`

**Comment by haysclark** [comment]
@vikr01 I hit an error when trying to run 'dev', is this something you are experiencing in your branch as well? Great job BTW! 

```
/Users/hays/.nvm/versions/node/v10.11.0/bin/node /Users/hays/.nvm/versions/node/v10.11.0/lib/node_modules/yarn/bin/yarn.js run dev
yarn run v1.10.0
$ cross-env START_HOT=1 node -r esm -r @babel/register ./internals/scripts/CheckPortInUse.js && cross-env START_HOT=1 yarn start-renderer-dev
$ cross-env NODE_ENV=development node --trace-warnings -r esm -r @babel/register ./node_modules/webpack-dev-server/bin/webpack-dev-server --config configs/webpack.config.renderer.dev.js
Starting Main Process...

> electron-react-boilerplate@0.16.0 start-main-dev /Users/hays/infinitedescent/electron-react-boilerplate
> cross-env HOT=1 NODE_ENV=development electron -r esm -r @babel/register ./app/main.dev.js

App threw an error during load
file:///Users/hays/infinitedescent/electron-react-boilerplate/node_modules/@babel/parser/lib/index.js:1
SyntaxError: /Users/hays/infinitedescent/electron-react-boilerplate/app/main.dev.js: 'return' outside of function (1:20)

> 1 | const _a69‍=exports;return _a69‍.r((function *(exports,require){"use strict";"main";let app,BrowserWindow;_a69‍.w("electron",[["app",["app"],function(v){app=v}],["BrowserWindow",["BrowserWindow"],function(v){BrowserWindow=v}]]);let MenuBuilder;_a69‍.w("./menu",[["default",["MenuBuilder"],function(v){MenuBuilder=v}]]);yield;/* eslint global-require: 0, flowtype-errors/show-errors: 0 */
    |                     ^
  2 | 
  3 | /**
  4 |  * This module executes inside of electron's main process. You can start
    at _class.raise (file:///Users/hays/electron-react-boilerplate/node_modules/@babel/parser/lib/index.js:3938:15)
    at _class.parseReturnStatement (file:///Users/hays/electron-react-boilerplate/node_modules/@babel/parser/lib/index.js:7488:12)
    at _class.parseStatementContent (file:///Users/hays/electron-react-boilerplate/node_modules/@babel/parser/lib/index.js:7178:21)
    at _class.parseStatement (file:///Users/hays/electron-react-boilerplate/node_modules/@babel/parser/lib/index.js:7144:17)
    at _class.parseStatement (file:///Users/hays/electron-react-boilerplate/node_modules/@babel/parser/lib/index.js:1902:57)
    at _class.parseBlockOrModuleBlockBody (file:///Users/hays/electron-react-boilerplate/node_modules/@babel/parser/lib/index.js:7695:23)
    at _class.parseBlockBody (file:///Users/hays/electron-react-boilerplate/node_modules/@babel/parser/lib/index.js:7682:10)
    at _class.parseTopLevel (file:///Users/hays/electron-react-boilerplate/node_modules/@babel/parser/lib/index.js:7109:10)
    at _class.parse (file:///Users/hays/electron-react-boilerplate/node_modules/@babel/parser/lib/index.js:8495:17)
    at parse (file:///Users/hays/electron-react-boilerplate/node_modules/@babel/parser/lib/index.js:10448:38)
```

**Comment by vikr01** [comment]
Thank you for the stacktrace @haysclark! Will look into this.

**Comment by vikr01** [comment]
@haysclark Please try out branch [`babel-7-migration-redo`](https://github.com/electron-react-boilerplate/electron-react-boilerplate/tree/babel-7-migration-redo) and let me know how this works for you.

**Comment by vikr01** [comment]
Closing in favor of #1898

**Pull request closed without merging**